rustls/src/client/tls13.rs

```diff
@@ -989,8 +992,8 @@ impl hs::State for ExpectFinished {
             let key = st
                 .key_schedule
                 .client_handshake_traffic_secret(
-                    &st.handshake
-                        .hash_at_client_recvd_server_hello,
+                    st.handshake
+                        .hash_at_client_recvd_server_hello.as_ref().unwrap(),
```

Comment:
I have a queued PR on top of this one to remove this `unwrap()`, by threading `hash_at_client_recvd_server_hello` through the handshake state machine directly.